EP-27: News Screen:

mordvintseva opened the pull request:
- Add NewsApi.kt
  - Add fun provideNewsPresenter(api: CoronavirusNewsApi) to AppModule
  - Update Constants.kt file
  - Update NetworkModule.kt to provide new api service
  - Create ui for news screen

mordvintseva committed:
EP-27: News Screen:

  - Add NewsApi.kt
  - Add fun provideNewsPresenter(api: CoronavirusNewsApi) to AppModule
  - Update Constants.kt file
  - Update NetworkModule.kt to provide new api service
  - Create ui for news screen

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsRecyclerViewAdapter.kt:
```diff
@@ -0,0 +1,60 @@
+            excerptTextView?.text = news.excerpt
+
+            val imageInfo: Image? = news.images?.firstOrNull()
+            if (imageInfo != null) {
```
let

Aliot-EI3 commented on app/src/main/res/layout/fragment_news.xml:
```diff
@@ -1,14 +1,29 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content" />
+
+        <!-- TODO: Update blank fragment layout -->
```
м?

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsView.kt:
```diff
@@ -0,0 +1,5 @@
+package com.example.epamcoronavirusmap.ui.news
+
+interface NewsView {
```
не верный нейминг интерфейса

Aliot-EI3 replied:
не имплементирует базовые

Aliot-EI3 replied:
перепиши на контракт. чтобы одинаковый вид был. к мише вопросы

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsFragment.kt:
```diff
@@ -1,25 +1,80 @@
+
+    private fun showProgress(show: Boolean) {
+        if (show) {
+            binding.progressBar.visibility = View.VISIBLE
```
binding. надо будет убрать скооперирутесь с мишей по этому поводу

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsFragment.kt:
```diff
@@ -1,25 +1,80 @@
+    }
+
+    private fun showProgress(show: Boolean) {
+        if (show) {
```
не красиво if написано. я показывал как. скопипащено - а нужно думать. исправь

Aliot-EI3 replied:
если не помнишь как - онбят спроси

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsPresenter.kt:
```diff
@@ -0,0 +1,48 @@
+    private lateinit var view: NewsView
+
+    init {
+        loadNews()
```
не здесь должна начинаться загрузка новостей

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsRecyclerViewAdapter.kt:
```diff
@@ -0,0 +1,60 @@
+        private var imageView: ImageView? = null
+
+        init {
+            titleTextView = view.findViewById(R.id.newsTitle)
```
findViewById - не нравится. по пробуй по другому. котлин помогает это упростить

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsPresenter.kt:
```diff
@@ -0,0 +1,48 @@
+
+class NewsPresenter(private val api: CoronavirusNewsApi) {
+
+    private var coroutineScope = CoroutineScope(Dispatchers.Main + Job())
```
надо будет переписать на rx. к мише. объяснит

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/news/NewsFragment.kt:
```diff
@@ -1,25 +1,80 @@
+        }
+    }
+
+    override fun update(result: NewsResult) {
```
не нравится нейминг

Aliot-EI3 replied:
+ не вьюха должен решать что делать с результатом, а презентр

mordvintseva committed:
EP-27: News screen

  - fragment_news.xml, NewsFragment.kt, NewsRecyclerViewAdapter.kt refactoring

mordvintseva committed:
Merge branches 'Feature/EP-27-News-Screen' and 'develop' of https://github.com/mishokU/EpamCoronavirusMap into Feature/EP-27-News-Screen

 Conflicts:
	app/build.gradle
	app/src/main/java/com/example/epamcoronavirusmap/api/Constants.kt
	app/src/main/java/com/example/epamcoronavirusmap/di/modules/AppModule.kt
	app/src/main/java/com/example/epamcoronavirusmap/di/modules/NetworkModule.kt

mordvintseva committed:
EP-27: News screen

  - Constants usage fix

mordvintseva committed:
EP-27: News screen

  - Create NewsContract.kt, apply changes to NewsFragment.kt, NewsPresenter.kt
  - Replace retrofit Call with Rx Observable in api

mordvintseva commented:
Поправила замечания

mordvintseva committed:
EP-27: News Screen

  - Update news api model

mordvintseva committed:
EP-27: News Screen

  - NewsFragment.kt: replace presenter.loadPosts method to onAttach
  - FragmentsBuilderModule.kt: remove contributeNewsFragment()

mordvintseva committed:
EP-27: News Screen

  - Remove findViewById method usage from NewsRecyclerViewAdapter.kt

mordvintseva committed:
Merge branches 'Feature/EP-27-News-Screen' and 'develop' of https://github.com/mishokU/EpamCoronavirusMap into Feature/EP-27-News-Screen

mordvintseva committed:
Merge branch 'develop' into Feature/EP-27-News-Screen
